### PR TITLE
Particle Rendering

### PR DESCRIPTION
**Pull request by fjz345**
Particle renders. Keep in mind that tasks that are not done yet:
More variables: rot/color/...
Creating/managing particleEffects.

### CODE REVIEW

**Review by AntonAsbrink — CHANGES_REQUESTED** [review]
Remove the testing code then it looks good.
Also be sure to fix sandbox activescene

File: StortSpelProjekt/Engine/src/Particles/ParticleSystem.cpp [hunk 1]
```diff
@@ -15,17 +19,26 @@ ParticleSystem& ParticleSystem::GetInstance()
...
+	{
+		//for(ParticleEffect& particleEffect : )
+		effect->Update(dt);
+		effect->updateResourceData();
+	}
+}
+
+void ParticleSystem::SetParticleEffect(ParticleEffect* effect)
+{
+	this->effect = effect;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Engine/src/Particles/ParticleSystem.cpp]: With correct code standard, 'this' is not necessary

File: StortSpelProjekt/Engine/src/Particles/ParticleSystem.h [hunk 2]
```diff
@@ -16,12 +17,16 @@ class ParticleSystem
...
+	void SetParticleEffect(ParticleEffect* effect);
+
+	void OnResetScene(); // Kanske, Jocke?
 
 private:
 	ParticleSystem();
 
-	ParticleEffect* effect;
+	// Temp code todo: vector of effectComponents. component::init() pushes in.
+	ParticleEffect* effect = nullptr;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Engine/src/Particles/ParticleSystem.h]: Code standard

File: StortSpelProjekt/Engine/src/Renderer/Renderer.cpp [hunk 3]
```diff
@@ -262,8 +267,22 @@ void Renderer::RenderUpdate(double dt)
...
 		ImGuiHandler::GetInstance().NewFrame();
 	}
+	
+	float3 right = m_pScenePrimaryCamera->GetRightVectorFloat3();
+	right.normalize();
+
+	float3 forward = m_pScenePrimaryCamera->GetDirectionFloat3();
+	forward.normalize();
+
+	// TODO: fix camera up vector
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Engine/src/Renderer/Renderer.cpp:277]: Camera does have up vector function

File: StortSpelProjekt/Engine/src/Renderer/Renderer.h [hunk 4]
```diff
@@ -126,6 +128,7 @@ class Renderer
 	void UnInitCameraComponent(component::CameraComponent* component);
 	void UnInitBoundingBoxComponent(component::BoundingBoxComponent* component);
 	void UnInitGUI2DComponent(component::GUI2DComponent* component);
+	void UnitParticleEmitterComponent(component::ParticleEmitterComponent* component);
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Engine/src/Renderer/Renderer.h]: Should this not be called 'UnInit...' instead of 'Unit...'?

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]

File: StortSpelProjekt/Engine/src/Particles/ParticleSystem.cpp (hunk 1 above)
Comment (fjz345) [reply StortSpelProjekt/Engine/src/Particles/ParticleSystem.cpp]: fixed

File: StortSpelProjekt/Engine/src/Particles/ParticleSystem.h (hunk 2 above)
Comment (fjz345) [reply StortSpelProjekt/Engine/src/Particles/ParticleSystem.h]: fixed

File: StortSpelProjekt/Engine/src/Renderer/Renderer.cpp (hunk 3 above)
Comment (fjz345) [reply StortSpelProjekt/Engine/src/Renderer/Renderer.cpp:277]: It is fuuuucked. It is always (0, 1, 0) when it should change once the camera changes. I guess the cameras up is world up.

File: StortSpelProjekt/Engine/src/Renderer/Renderer.h (hunk 4 above)
Comment (fjz345) [reply StortSpelProjekt/Engine/src/Renderer/Renderer.h]: Good catch, just noticed myself ^^.

**Review by LeoWikstrom — DISMISSED** [review]

**Review by LeoWikstrom — APPROVED** [review]

**Review by AntonAsbrink — APPROVED** [review]

**Pull request merged**